### PR TITLE
Update legal.md

### PR DESCRIPTION
**Pull request by thehale**
On July 31, 2023 @abbycabs asked me to review the legal article on Open Source Guides for tweaks and improvements. 

This PR...
- Simplifies the legalese in a few places.
- Adds more links to licenses and data behind various claims.
- Adds more descriptions for different classes of licenses.
- Improves internal linking within the article.

For now, I'm keeping this in draft status to (1) allow for more discussion about the changes and (2) give time to flesh out the new litigation section (if we decide to pursue it).

**Checklist**
- [X] Have you followed the [contributing guidelines](https://github.com/github/opensource.guide/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you explained what your changes do, and why they add value to the Guides?

**Please note: we will close your PR without comment if you do not check the boxes above and provide ALL requested information.**

-----

### CODE REVIEW

File: _articles/legal.md
Comment (thehale) [inline _articles/legal.md:15]: This article does not and cannot "cover the reader's legal needs". Rather it is a broad introduction to the legal side of OSS.

File: _articles/legal.md
Comment (thehale) [inline _articles/legal.md:25]: I felt like this paragraph came across as "too legalese". I tried to improve the flow from the previous paragraph and simplify the language without removing any critical ideas.

File: _articles/legal.md
Comment (thehale) [inline _articles/legal.md:43]: Added a source for license popularities (libraries.io by Tidelift).

File: _articles/legal.md
Comment (thehale) [inline _articles/legal.md:57]: As I understand it, the goal of this first paragraph is to *quickly* give readers the name of license they can use and move on with their lives if they are tired of thinking about legal stuff.

This rephrasing puts the name of MIT in the first line instead of the second.

File: _articles/legal.md
Comment (thehale) [inline _articles/legal.md:66]: The long parentheticals defining each class of license break the reader's concentration. By re-writing in bullet point form, there's more space to explain without creating a wall of text, *and* it matches the layout of the following *communities* section.

I also added details about "limited copyleft" licenses to bring greater awareness to their unique properties. I deliberately chose the term "limited copyleft" to emphasize the limited reach of their copyleft provisions (the distinguishing characteristic of these licenses). The more common term is "weak copyleft", but I have seen that term prompt faulty impressions that such licenses somehow only provide "weak/less-enforceable" copyright protections.

File: _articles/legal.md
Comment (thehale) [inline _articles/legal.md:68]: Adding a use case for choosing a "limited copyleft" license.

File: _articles/legal.md
Comment (thehale) [inline _articles/legal.md:71]: Emphasizing that the terms of "limited copyleft" licenses, particularly the MPL 2.0, are compatible with usage in proprietary software (a common misunderstanding).

File: _articles/legal.md
Comment (thehale) [inline _articles/legal.md:168]: The linked Maintainer Community discussion highlights what I consider this article's biggest gap -- how to handle infringement.

The obvious answer is "get a lawyer", but I think it would be helpful to provide links to OSS-specific foundations/communities that help with litigation.

**Comment by abbycabs** [comment]
Tagging @karasowles while I'm on leave! I believe she scheduled a followup call with all the writers.

File: _articles/legal.md
Comment (thehale) [inline _articles/legal.md:87]: I have a project which helps with identifying the existing copyright holders in your code, but I don't want to include it without prior discussion to avoid "conflicts of interest"/"self-promotion" concerns.

https://github.com/thehale/git-authorship

Comment (ahpook) [reply _articles/legal.md:87]: I don't think this would present a conflict of interest, if you still want to include it please feel free to do so!

**Review by ahpook — APPROVED** [review]
Hi @thehale, sorry this PR has languished - It didn't register on our automation. This all looks great and I appreciate your commentary on why you made the edits you did. You mentioned adding a reference to a copyright/attribution project, do you want to add that inline before merge? If so, go ahead and do so - if not, I'll merge this as-is. Thank you!

**Comment by thehale** [comment]
Thanks! I'll add the link by this weekend.

File: _articles/legal.md
Comment (mlinksva) [inline _articles/legal.md:15]: I wouldn't call this a legal course, here's another idea:

```suggestion
Sharing your creative work with the world can be an exciting and rewarding experience. It can also mean a bunch of legal things you didn't know you had to worry about. Thankfully, with this guide you don't have to start from scratch. (Before you dig in, be sure to read our [disclaimer](/notices/).)
```

File: _articles/legal.md
Comment (mlinksva) [inline _articles/legal.md:23]: I don't love this because it sounds like creating a document is what the user needs to do, which is definitely not the case. Here's an alternative:
```suggestion
Open source is an unusual circumstance, however, because the author expects that others will use, modify, and share the work. But because the legal default is still exclusive copyright, you need to explicitly give these permissions with a license.
```

File: _articles/legal.md
Comment (mlinksva) [inline _articles/legal.md:43]: libraries.io's stats (which are probably dominated by npm) don't support the statement! Subbing in a new resource that does, and remove "the most" to make it more likely to be true forever.
```suggestion
[MIT](https://choosealicense.com/licenses/mit/), [Apache 2.0](https://choosealicense.com/licenses/apache-2.0/), and [GPLv3](https://choosealicense.com/licenses/gpl-3.0/) are [popular open source licenses](https://innovationgraph.github.com/global-metrics/licenses), but there are other options to choose from. You can find the full text of these licenses, and instructions on how to use them, on [choosealicense.com](https://choosealicense.com/).
```

File: _articles/legal.md
Comment (mlinksva) [reply _articles/legal.md:66]: Get the parens aren't ideal but feel like the bullets make it less obvious what the connection between dependencies' licenses and your project are. And I don't think it's really necessary to introduce the concept of limited or weak copyleft at this point.

File: _articles/legal.md
Comment (mlinksva) [inline _articles/legal.md:70]: I don't like this change. The ability to use in proprietary software is also true of MIT, and is only one of the reasons permissive is preferred. I again also don't see any reason to introduce limited copyleft here.

I'd also tone down the original recommendation a bit; the years since this was written have shown even more that MIT's implied patent license is very rarely if ever a blocker. Here's a suggestion:

```suggestion
* **Do you want your project to appeal to large businesses?** A large business may be comforted by an express patent license from all contributors. In this case, [Apache 2.0](https://choosealicense.com/licenses/apache-2.0/) has you (and them) covered.
```

**Review by mlinksva — CHANGES_REQUESTED** [review]
Thanks for the tweaks (I wrote most of the original text and yeah do overuse parentheticals)! Misc tweak-tweaks.

File: _articles/legal.md
Comment (mlinksva) [reply _articles/legal.md:71]: Missed this in my initial review. I'd also skip this. Again, limited copyleft isn't a concept that needs to be in this introductory material, and the edit isn't true except for file-based copyleft like MPL-2.0 (from experience, LGPL always requires closer scrutiny), and file-based copyleft is really not a concept that needs to be delved into here.

**Comment by ahpook** [comment]
Thanks for the update @thehale. Looks like you got a few helpful improvements from @mlinksva code review - care to address those, and we'll merge this in?

**Comment by thehale** [comment]
@ahpook, Yep I saw the great feedback from @mlinksva, but it will be a least few days before I can incorporate his recommendations. Hoping to have it done by this weekend.

File: _articles/legal.md
Comment (WebSnke) [inline _articles/legal.md:89]: I suggest putting "see below" in brackets here too as you did below.

```suggestion
Alternatively, you can have contributors agree in advance via an additional contributor agreement ([see below](#does-my-project-need-an-additional-contributor-agreement)) to certain license changes under certain conditions, beyond those allowed by your existing open source license. This shifts the complexity of changing licenses a bit. You'll need more help from your lawyers up front, and you'll still want to clearly communicate with your project's stakeholders when executing a license change.
```

**Comment by ahpook** [comment]
@thehale it's a friendly Open Source Friday ping on this PR! Looks like all the comments have suggestions that would be straightforward to batch-commit - could you click through them and update your branch? We can iterate further in the future but I'd love to get this merged and closed out.  🚀 ❤️

File: _articles/legal.md
Comment (thehale) [reply _articles/legal.md:68]: Good feedback. My edits shift the focus back towards the ramifications of a dependency's license on the larger project.

I still think it's helpful to briefly distinguish between strong/limited copyleft licenses since their requirements for downstream project licensing differ so widely. That was a big point of confusion for me when I started out in open source. Additionally, including the distinction here prepares the reader to understand the discussion a few paragraphs later of why some companies want a "strong" copyleft license.

The linked article about the "additional rules" of limited copyleft licenses is from FOSSA, a blog about open-source licensing. I am not affiliated with them, just a fan of their detail and approachability.

File: _articles/legal.md
Comment (thehale) [reply _articles/legal.md]: Fair enough. Upon re-reading I'm also recognizing that this part of the article is best if it recommends a single concrete license for each situation, and the Apache-2.0 is certainly the industry's preferred choice for explicit patent grants.

File: _articles/legal.md
Comment (thehale) [reply _articles/legal.md]: Originally, I only linked up the existing text. In the latest edits, I took your re-write a few steps further to remove redundant clauses.

e.g. "under certain conditions" and "beyond those allowed ..." are already implied by the need for a separate contract.

**Comment by thehale** [comment]
> @thehale it's a friendly Open Source Friday ping on this PR! 

Thanks for the follow up @ahpook. Typically, my open-source work is limited to irregular weekends, so tonight was the first opportunity to dig back into the feedback. Hopefully this latest set of edits covers the great notes made!

**Review by ahpook — APPROVED** [review]
This looks great! Thank you @thehale and everyone who commented and gave feedback.

**Comment by ahpook** [comment]
I'm going to go ahead and merge this, if there are future iterations we want to make on this doc let's do so in a fresh PR. Cheers all 🚀

**Pull request merged**
